### ISSUE TITLE
"module Ecto.DateTimeWithTimezone is not available" when casting DateTimeWithTimezone with a castable binary

### ISSUE THREAD

**Issue by tessi** [bitwalker/timex_ecto#29]
First, thanks for timex_ecto ❤️ 

I (as a relative elixir noob) found some behaviour which I think might be a bug:
Casting ISO8601 encoded (or any other encoded date) date-strings results in an error being thrown instead of a casted `DateTime`.

Doing

``` elixir
Timex.Ecto.DateTimeWithTimezone.cast("2016-02-14T12:34:00.321+00:00")
```

fails with 

```
iex(1)> Timex.Ecto.DateTimeWithTimezone.cast("2016-02-14T12:34:00.321+00:00")
** (UndefinedFunctionError) undefined function Ecto.DateTimeWithTimezone.cast/1 (module Ecto.DateTimeWithTimezone is not available)
                 Ecto.DateTimeWithTimezone.cast("2016-02-14T12:34:00.321+00:00")
    (timex_ecto) lib/types/datetimetz.ex:57: Timex.Ecto.DateTimeWithTimezone.cast/1
```

The error seems to origin from [this line in our codebase](https://github.com/bitwalker/timex_ecto/blob/a42ff1584306ef92746fa6ffe97c3bd74f8a34d6/lib/types/datetimetz.ex#L57). It tries to call `Ecto.DateTimeWithTimezone.cast`. I cannot find a definition for `Ecto.DateTimeWithTimezone` (including in our dependencies).

How was this supposed to work originally? I'd be glad to help finding a fix.

**Comment by tessi** [comment]
I've added [a pull request](https://github.com/bitwalker/timex_ecto/pull/30) adding tests to `DateTimeWithTimezone` -- they currently fail (on purpose) because of this bug.

**Comment by bitwalker** [comment]
This has been addressed and a new release (1.1.1) pushed to hex.

**Comment by tessi** [comment]
\o/ thanks
